pkg/engine/resolve/resolve.go

```diff
@@ -237,7 +250,7 @@ type Fetch interface {
 type Fetches []Fetch
 
 type DataSource interface {
-	Load(ctx context.Context, input []byte, bufPair *BufPair) (err error)
+	Load(ctx context.Context, input []byte) (data []byte, err error)
```

Comment:
I don't like returning byte slice, I'd much rather use io.Writer as argument and return err only